### PR TITLE
docs: Add lazydocs test & initial markdowns

### LINKED ISSUE #100: Auto generate API documentation and publish them to documentation site
Documentation of the module (methods, classes...) should be executed, using `lazydocs`. 

The following things should be executed

- [ ] Document all methods and modules
- [ ] Run lazydocs to generate documentation files, from a unit test
- [ ] Ensure an overview (`index.md`) is generated as well

### PR DESCRIPTION
**Pull request by SamVanhoutte**
Closes #100

### CODE REVIEW

**Comment by SamVanhoutte** [comment]
@Mcalewaert - please let me know what is causing the netlify actions to fail.  I can update the generation, as I already did for the non-closed img tags

**Comment by Mcalewaert** [comment]
> @Mcalewaert - please let me know what is causing the netlify actions to fail. I can update the generation, as I already did for the non-closed img tags

"Fixed" together with @SamVanhoutte

File: build/ci-build.yml [hunk 1]
```diff
@@ -79,10 +81,10 @@ steps:
...
     pythonUploadServiceConnection: 'Arcus ML PyPi feed'
 
-- script: |
-   pdoc --html --force --config show_source_code=False -o docs/ arcus/ml 
-  displayName: 'Generate code documentation'
-  continueOnError: true
+# - script: |
+#    pdoc --html --force --config show_source_code=False -o docs/ arcus/ml 
+#   displayName: 'Generate code documentation'
+#   continueOnError: true
```
Comment (stijnmoreels) [inline build/ci-build.yml:87]: I guess this can be removed now?

**Review by stijnmoreels — APPROVED** [review]
This is a good start to have an API reference of the library.  Great stuff! 👍 This is probably not enough for a full feature documentation as it lacks examples.

File: build/ci-build.yml (hunk 1 above)
Comment (SamVanhoutte) [reply build/ci-build.yml:87]: yes , but we still have to make sure the generated md files are compatible & can be parsed nicely with the netlify docs

**Comment by SamVanhoutte** [comment]
> 

@stijnmoreels : samples can normally be added in the docstring.  just have to look at the parsing & generation of the output

File: build/ci-build.yml (hunk 1 above)
Comment (SamVanhoutte) [reply build/ci-build.yml]: removed with new commit

**Comment by stijnmoreels** [comment]
> > 
> 
> @stijnmoreels : samples can normally be added in the docstring. just have to look at the parsing & generation of the output

Yeah, I also meant maybe with Jupiter notebooks, graphics... whatever to make it a 'real' example. Those kind of code samples, directly in the code, will maybe only pollute the code.

**Pull request merged**